library/src/main/java/com/climate/mirage/Mirage.java

```diff
@@ -251,13 +272,16 @@ public MirageTask go(MirageRequest request) {
             return null;
         }
 
+        // TODO: clean up this duplicate
         if (request.memoryCache() == null) request.memoryCache(defaultMemoryCache);
         if (request.diskCache() == null) request.diskCache(defaultDiskCache);
         if (request.executor() == null) request.executor(defaultExecutor);
         if (request.urlFactory() == null) request.urlFactory(defaultUrlConnectionFactory);
 
         // if the url is blank, fault out immediately
-        if (request.uri() == null || TextUtils.isEmpty(request.uri().toString())) {
+        // TODO: clean up this duplicate
```

Comment:
is this from work, or still work to do?

Reply:
this is duplication that was already there that i'm calling out now. There's a good bit of stuff I'd clean up, but didn't want to hold up partha any longer. I can slow down and add more docs and do some clean up if you think it's worth while.